src/disco/reducers/discoResults.js

```diff
@@ -63,26 +66,52 @@ export function getDiscoResults({
   };
 }
 
-type LoadDiscoResultsParams = ExternalDiscoResultsType;
+type LoadDiscoResultsParams = {|
+  results: Array<ExternalDiscoResultType>,
+|};
 
 type LoadDiscoResultsAction = {|
   type: typeof LOAD_DISCO_RESULTS,
   payload: LoadDiscoResultsParams,
 |};
 
 export function loadDiscoResults({
-  entities,
-  result,
+  results,
 }: LoadDiscoResultsParams = {}): LoadDiscoResultsAction {
-  invariant(entities, 'entities are required');
-  invariant(result, 'result is required');
+  invariant(results, 'results are required');
 
   return {
     type: LOAD_DISCO_RESULTS,
-    payload: { entities, result },
+    payload: { results },
   };
 }
 
+export const createInternalResult = (
+  result: ExternalDiscoResultType,
+): DiscoResultType => {
+  return {
+    addonId: result.addon.id,
+    description: result.description || null,
+    heading: result.heading,
+    isRecommendation: result.is_recommendation,
+  };
+};
+
+type CreateExternalAddonMapParams = {|
+  results: Array<ExternalDiscoResultType>,
+|};
+
+export const createExternalAddonMap = ({
+  results,
+}: CreateExternalAddonMapParams): ExternalAddonMap => {
```

Comment:
It's unfortunate to say that this returns `ExternalAddonMap` because the `addon` objects are missing a lot of properties. I know that if you type it correctly then Flow will explode so I'm not sure what to do. I worry that something in Discopane will break one day because it's relying on a non-existent property (such as `addon.description` or whatever).